### ISSUE TITLE
Add oelint-adv support

### ISSUE THREAD

**Issue by jclsn** [Freed-Wu/bitbake-language-server#6]
Would be nice to have support for oelint-adv:

https://github.com/priv-kweihmann/oelint-adv

**Comment by Freed-Wu** [comment]
![screen-2024-03-01-15-45-12](https://github.com/Freed-Wu/bitbake-language-server/assets/32936898/455f9b9c-1985-4b20-aae0-ebc1ead1f7d9)

**Comment by jclsn** [comment]
Thanks, still a bit unstable it seems. It crashes when I am typing

```
[coc.nvim] Connection to server bitbake is erroring, Header must provide a Content-Length property.
{"{\"params\"":"{\"type\": 3, \"message\": \"Add /home/jan/Workspace/Yocto/meta-something/recipes-something/my-recipes/my-recipe.bb\"}, \"method\": \"window/showMessage\", \"jsonrpc\": \"2.0\"}Loaded rules:\n\toelint.append.protvars\n\toelint.append.protvars.LICENSE\n\toelint.append.protvars.LI
C_FILES_CHKSUM\n\toelint.append.protvars.PR\n\toelint.append.protvars.PV\n\toelint.append.protvars.SRCREV\n\toelint.bbclass.underscores\n\toelint.exportfunction.dash\n\toelint.file.inactiveupstreamdetails\n\toelint.file.inappropriatemsg\n\toelint.file.includenotfound\n\toelint.file.includerelpa
th\n\toelint.file.inlinesuppress_na\n\toelint.file.nospaces\n\toelint.file.patchsignedoff\n\toelint.file.requireinclude\n\toelint.file.requirenotfound\n\toelint.file.underscores\n\toelint.file.upstreamstatus\n\toelint.func.specific\n\toelint.newline.consecutive\n\toelint.newline.eof\n\toelint.s
paces.emptyline\n\toelint.spaces.linebeginning\n\toelint.spaces.linecont\n\toelint.spaces.lineend\n\toelint.tabs.notabs\n\toelint.task.addnotaskbody\n\toelint.task.customorder\n\toelint.task.dash\n\toelint.task.docstrings\n\toelint.task.heredocs\n\toelint.task.multifragments\n\toelint.task.noan
onpython\n\toelint.task.nocopy\n\toelint.task.nomkdir\n\toelint.task.nopythonprefix\n\toelint.task.order\n\toelint.task.order.do_build\n\toelint.task.order.do_compile\n\toelint.task.order.do_configure\n\toelint.task.order.do_fetch\n\toelint.task.order.do_install\n\toelint.task.order.do_package\
n\toelint.task.order.do_patch\n\toelint.task.order.do_populate_sysroot\n\toelint.task.order.do_unpack\n\toelint.task.pythonprefix\n\toelint.var.addpylib\n\toelint.var.bbclassextend\n\toelint.var.filesoverride\n\toelint.var.improperinherit\n\toelint.var.licenseremotefile\n\toelint.var.mandatoryv
ar\n\toelint.var.mandatoryvar.DESCRIPTION\n\toelint.var.mandatoryvar.HOMEPAGE\n\toelint.var.mandatoryvar.LICENSE\n\toelint.var.mandatoryvar.SRC_URI\n\toelint.var.mandatoryvar.SUMMARY\n\toelint.var.multiinclude\n\toelint.var.multiinherit\n\toelint.var.nativefilename\n\toelint.var.nativesdkfilena
me\n\toelint.var.order\n\toelint.var.order.AUTHOR\n\toelint.var.order.BBCLASSEXTEND\n\toelint.var.order.BUGTRACKER\n\toelint.var.order.DEPENDS\n\toelint.var.order.DESCRIPTION\n\toelint.var.order.EXTRA_OECONF\n\toelint.var.order.EXTRA_QMAKEVARS_POST\n\toelint.var.order.FILES\n\toelint.var.order.
HOMEPAGE\n\toelint.var.order.LICENSE\n\toelint.var.order.LIC_FILES_CHKSUM\n\toelint.var.order.PACKAGECONFIG\n\toelint.var.order.PACKAGES\n\toelint.var.order.PACKAGE_ARCH\n\toelint.var.order.PROVIDES\n\toelint.var.order.PV\n\toelint.var.order.RCONFLICTS\n\toelint.var.order.RDEPENDS\n\toelint.var
.order.RPROVIDES\n\toelint.var.order.RRECOMMENDS\n\toelint.var.order.RSUGGESTS\n\toelint.var.order.S\n\toelint.var.order.SECTION\n\toelint.var.order.SRCREV\n\toelint.var.order.SRC_URI\n\toelint.var.order.SUMMARY\n\toelint.var.order.inherit\n\toelint.var.override\n\toelint.var.rootfspostcmd\n\to
elint.var.srcuriwildcard\n\toelint.var.suggestedvar\n\toelint.var.suggestedvar.BBCLASSEXTEND\n\toelint.var.suggestedvar.BUGTRACKER\n\toelint.var.suggestedvar.CVE_PRODUCT\n\toelint.var.suggestedvar.SECTION\n\toelint.vars.appendop\n\toelint.vars.autorev\n\toelint.vars.bbvars\n\toelint.vars.bbvars
.BBDEBUG\n\toelint.vars.bbvars.BBFILES\n\toelint.vars.bbvars.BBFILE_COLLECTIONS\n\toelint.vars.bbvars.BBFILE_PATTERN\n\toelint.vars.bbvars.BBFILE_PRIORITY\n\toelint.vars.bbvars.BBINCLUDED\n\toelint.vars.bbvars.BBINCLUDELOGS\n\toelint.vars.bbvars.BBINCLUDELOGS_LINES\n\toelint.vars.bbvars.BBLAYER
S\n\toelint.vars.bbvars.BBMASK\n\toelint.vars.bbvars.BBPATH\n\toelint.vars.bbvars.BBSERVER\n\toelint.vars.bbvars.BBVERSIONS\n\toelint.vars.bbvars.BB_CONSOLELOG\n\toelint.vars.bbvars.BB_CURRENTTASK\n\toelint.vars.bbvars.BB_DANGLINGAPPENDS_WARNONLY\n\toelint.vars.bbvars.BB_DEFAULT_TASK\n\toelint.
vars.bbvars.BB_DISKMON_DIRS\n\toelint.vars.bbvars.BB_DISKMON_WARNINTERVAL\n\toelint.vars.bbvars.BB_ENV_EXTRAWHITE\n\toelint.vars.bbvars.BB_ENV_WHITELIST\n\toelint.vars.bbvars.BB_FETCH_PREMIRRORONLY\n\toelint.vars.bbvars.BB_FILENAME\n\toelint.vars.bbvars.BB_GENERATE_MIRROR_TARBALLS\n\toelint.var
s.bbvars.BB_HASHBASE_WHITELIST\n\toelint.vars.bbvars.BB_HASHCHECK_FUNCTION\n\toelint.vars.bbvars.BB_HASHCONFIG_WHITELIST\n\toelint.vars.bbvars.BB_INVALIDCONF\n\toelint.vars.bbvars.BB_LOGFMT\n\toelint.vars.bbvars.BB_NICE_LEVEL\n\toelint.vars.bbvars.BB_NO_NETWORK\n\toelint.vars.bbvars.BB_NUMBER_P
ARSE_THREADS\n\toelint.vars.bbvars.BB_NUMBER_THREADS\n\toelint.vars.bbvars.BB_ORIGENV\n\toelint.vars.bbvars.BB_PRESERVE_ENV\n\toelint.vars.bbvars.BB_RUNFMT\n\toelint.vars.bbvars.BB_RUNTASK\n\toelint.vars.bbvars.BB_SCHEDULER\n\toelint.vars.bbvars.BB_SCHEDULERS\n\toelint.vars.bbvars.BB_SETSCENE_D
EPVALID\n\toelint.vars.bbvars.BB_SETSCENE_VERIFY_FUNCTION\n\toelint.vars.bbvars.BB_SIGNATURE_EXCLUDE_FLAGS\n\toelint.vars.bbvars.BB_SIGNATURE_HANDLER\n\toelint.vars.bbvars.BB_SRCREV_POLICY\n\toelint.vars.bbvars.BB_STAMP_POLICY\n\toelint.vars.bbvars.BB_STAMP_WHITELIST\n\toelint.vars.bbvars.BB_ST
RICT_CHECKSUM\n\toelint.vars.bbvars.BB_TASKHASH\n\toelint.vars.bbvars.BB_TASK_NICE_LEVEL\n\toelint.vars.bbvars.BB_VERBOSE_LOGS\n\toelint.vars.bbvars.BB_WORKERCONTEXT\n\toelint.vars.bbvars.BITBAKE_UI\n\toelint.vars.bbvars.BUILDNAME\n\toelint.vars.bbvars.CACHE\n\toelint.vars.bbvars.DL_DIR\n\toeli
nt.vars.bbvars.FILE\n\toelint.vars.bbvars.FILESDIR\n\toelint.vars.bbvars.FILESPATH\n\toelint.vars.bbvars.INHERIT\n\toelint.vars.bbvars.LAYERDEPENDS\n\toelint.vars.bbvars.LAYERDIR\n\toelint.vars.bbvars.LAYERVERSION\n\toelint.vars.bbvars.MIRRORS\n\toelint.vars.bbvars.MULTI_PROVIDER_WHITELIST\n\to
elint.vars.bbvars.PERSISTENT_DIR\n\toelint.vars.bbvars.PREFERRED_PROVIDER\n\toelint.vars.bbvars.PREFERRED_PROVIDERS\n\toelint.vars.bbvars.PREFERRED_VERSION\n\toelint.vars.bbvars.PREMIRRORS\n\toelint.vars.bbvars.PRSERV_HOST\n\toelint.vars.bbvars.STAMP\n\toelint.vars.bbvars.TOPDIR\n\toelint.vars.
bugtrackerisurl\n\toelint.vars.dependsappend\n\toelint.vars.dependsclass\n\toelint.vars.dependsordered\n\toelint.vars.descriptionsame\n\toelint.vars.descriptiontoobrief\n\toelint.vars.doublemodify\n\toelint.vars.downloadfilename\n\toelint.vars.duplicate\n\toelint.vars.dusageinpkgfuncs\n\toelint
.vars.fileextrapaths\n\toelint.vars.fileextrapathsop\n\toelint.vars.filessetting\n\toelint.vars.filessetting.double\n\toelint.vars.filessetting.hidden\n\toelint.vars.homepageping\n\toelint.vars.homepageprefix\n\toelint.vars.inconspaces\n\toelint.vars.insaneskip\n\toelint.vars.licensesdpx\n\toel
int.vars.licfileprefix\n\toelint.vars.listappend\n\toelint.vars.mispell\n\toelint.vars.mispell.unknown\n\toelint.vars.multilineident\n\toelint.vars.notneededspace\n\toelint.vars.notrailingslash\n\toelint.vars.overrideappend\n\toelint.vars.pathhardcode\n\toelint.vars.pathhardcode.bindir\n\toelin
t.vars.pathhardcode.datadir\n\toelint.vars.pathhardcode.docdir\n\toelint.vars.pathhardcode.includedir\n\toelint.vars.pathhardcode.infodir\n\toelint.vars.pathhardcode.libdir\n\toelint.vars.pathhardcode.libexecdir\n\toelint.vars.pathhardcode.localstatedir\n\toelint.vars.pathhardcode.mandir\n\toel
int.vars.pathhardcode.nonarch_base_libdir\n\toelint.vars.pathhardcode.sbindir\n\toelint.vars.pathhardcode.servicedir\n\toelint.vars.pathhardcode.sharedstatedir\n\toelint.vars.pathhardcode.sysconfdir\n\toelint.vars.pathhardcode.systemd_system_unitdir\n\toelint.vars.pathhardcode.systemd_unitdir\n
\toelint.vars.pathhardcode.systemd_user_unitdir\n\toelint.vars.pbpusage\n\toelint.vars.pkgspecific\n\toelint.vars.pkgspecific.ALLOW_EMPTY\n\toelint.vars.pkgspecific.FILES\n\toelint.vars.pkgspecific.RCONFLICTS\n\toelint.vars.pkgspecific.RDEPENDS\n\toelint.vars.pkgspecific.RPROVIDES\n\toelint.var
s.pkgspecific.RRECOMMENDS\n\toelint.vars.pkgspecific.RREPLACES\n\toelint.vars.pkgspecific.RSUGGESTS\n\toelint.vars.pkgspecific.pkg_postinst\n\toelint.vars.pkgspecific.pkg_postrm\n\toelint.vars.pkgspecific.pkg_preinst\n\toelint.vars.pkgspecific.pkg_prerm\n\toelint.vars.pnbpnusage\n\toelint.vars.
pnusagediscouraged\n\toelint.vars.sectionlowercase\n\toelint.vars.spacesassignment\n\toelint.vars.specific\n\toelint.vars.srcuriappend\n\toelint.vars.srcurichecksum\n\toelint.vars.srcuridomains\n\toelint.vars.srcurifile\n\toelint.vars.srcurigittag\n\toelint.vars.srcurioptions\n\toelint.vars.src
urisrcrevtag\n\toelint.vars.summary80chars\n\toelint.vars.summarylinebreaks\n\toelint.vars.valuequoted\n\toelint.vars.virtual\nParsing /home/jan/Workspace/Yocto/meta-aerq-lros/recipes-lros/lru-ledbus/lru-ledbus_0.1.bb\nParsing /home/jan/Workspace/Yocto/meta-aerq-lros/classes/install-to.bbclass\
[coc.nvim] Connection to server bitbake is erroring, Header must provide a Content-Length property.
{"{\"params\"":"{\"uri\": \"file:///home/jan/Workspace/Yocto/meta-something/recipes-something/my-recipes/my-recipe.bb\", \"diagnostics\": [{\"range\": {\"start\": {\"line\": 0, \"character\": 0}, \"end\": {\"line\": 1, \"character\": 0}}, \"message\": \"bbclass filenames should not contain '-'.
 Replace it by '_'\", \"severity\": 1}, {\"range\": {\"start\": {\"line\": 2, \"character\": 0}, \"end\": {\"line\": 3, \"character\": 0}}, \"message\": \"Don't use tabs use spaces\", \"severity\": 2}, {\"range\": {\"start\": {\"line\": 2, \"character\": 0}, \"end\": {\"line\": 3, \"character\"
: 0}}, \"message\": \"Every custom task should have a doc string set by task[doc] = ''\", \"severity\": 3}, {\"range\": {\"start\": {\"line\": 7, \"character\": 0}, \"end\": {\"line\": 8, \"character\": 0}}, \"message\": \"Don't use tabs use spaces\", \"severity\": 2}, {\"range\": {\"start\": {
\"line\": 7, \"character\": 0}, \"end\": {\"line\": 8, \"character\": 0}}, \"message\": \"Every custom task should have a doc string set by task[doc] = ''\", \"severity\": 3}, {\"range\": {\"start\": {\"line\": 11, \"character\": 0}, \"end\": {\"line\": 12, \"character\": 0}}, \"message\": \"Do
n't use tabs use spaces\", \"severity\": 2}, {\"range\": {\"start\": {\"line\": 11, \"character\": 0}, \"end\": {\"line\": 12, \"character\": 0}}, \"message\": \"Every custom task should have a doc string set by task[doc] = ''\", \"severity\": 3}, {\"range\": {\"start\": {\"line\": 15, \"charac
ter\": 0}, \"end\": {\"line\": 16, \"character\": 0}}, \"message\": \"Every custom task should have a doc string set by task[doc] = ''\", \"severity\": 3}, {\"range\": {\"start\": {\"line\": 15, \"character\": 0}, \"end\": {\"line\": 16, \"character\": 0}}, \"message\": \"Don't use tabs use spa
ces\", \"severity\": 2}, {\"range\": {\"start\": {\"line\": 19, \"character\": 0}, \"end\": {\"line\": 20, \"character\": 0}}, \"message\": \"Don't use tabs use spaces\", \"severity\": 2}, {\"range\": {\"start\": {\"line\": 19, \"character\": 0}, \"end\": {\"line\": 20, \"character\": 0}}, \"me
ssage\": \"Every custom task should have a doc string set by task[doc] = ''\", \"severity\": 3}, {\"range\": {\"start\": {\"line\": 23, \"character\": 0}, \"end\": {\"line\": 24, \"character\": 0}}, \"message\": \"Don't use tabs use spaces\", \"severity\": 2}, {\"range\": {\"start\": {\"line\":
 23, \"character\": 0}, \"end\": {\"line\": 24, \"character\": 0}}, \"message\": \"Every custom task should have a doc string set by task[doc] = ''\", \"severity\": 3}, {\"range\": {\"start\": {\"line\": 31, \"character\": 0}, \"end\": {\"line\": 32, \"character\": 0}}, \"message\": \"Don't use
 tabs use spaces\", \"severity\": 2}, {\"range\": {\"start\": {\"line\": 31, \"character\": 0}, \"end\": {\"line\": 32, \"character\": 0}}, \"message\": \"Every custom task should have a doc string set by task[doc] = ''\", \"severity\": 3}, {\"range\": {\"start\": {\"line\": 42, \"character\": 
0}, \"end\": {\"line\": 43, \"character\": 0}}, \"message\": \"Don't use tabs use spaces\", \"severity\": 2}, {\"range\": {\"start\": {\"line\": 42, \"character\": 0}, \"end\": {\"line\": 43, \"character\": 0}}, \"message\": \"Every custom task should have a doc string set by task[doc] = ''\", 
\"severity\": 3}, {\"range\": {\"start\": {\"line\": 0, \"character\": 0}, \"end\": {\"line\": 1, \"character\": 0}}, \"message\": \"Variable 'BUGTRACKER' should be set\", \"severity\": 3}, {\"range\": {\"start\": {\"line\": 0, \"character\": 0}, \"end\": {\"line\": 1, \"character\": 0}}, \"mes
sage\": \"Variable 'SECTION' should be set\", \"severity\": 3}, {\"range\": {\"start\": {\"line\": 0, \"character\": 0}, \"end\": {\"line\": 1, \"character\": 0}}, \"message\": \"BBCLASSEXTEND should be set if possible\", \"severity\": 3}, {\"range\": {\"start\": {\"line\": 0, \"character\": 0}
[coc.nvim] Connection to server bitbake is erroring, Header must provide a Content-Length property.
{"{\"id\"":"3, \"jsonrpc\": \"2.0\", \"result\": [{\"range\": {\"start\": {\"line\": 33, \"character\": 22}, \"end\": {\"line\": 33, \"character\": 32}}, \"target\": \"file:///home/jan/Workspace/Yocto/meta-something/recipes-something/my-recipes/my-recipe.bb\"}]}Content-Length: 43","content-type":"application/
vscode-jsonrpc; charset=utf-8"}. Shutting down server.
```

**Comment by jclsn** [comment]
You should probably keep it disabled by default and add the possibility to enable it to the configuration page. It also needs to find a config file in `HOME` or the project root. It think the default name for the config file is `.oelint.cfg` and is in .ini format. 

https://github.com/priv-kweihmann/oelint-adv/blob/master/docs/.oelint.cfg.example

Seems to work if you set it as environment variable ``OELINT_CONFIG``

**Comment by Freed-Wu** [comment]
1. What is the version?
2. Can you share a reproduce steps, include some test files?

**Comment by jclsn** [comment]
1. Version of what? I am using Vim with the [CoC plugin](https://github.com/neoclide/coc.nvim)
3. You can test the parsing with basically any Bitbake recipe (.bb files) in [openembedded-core](https://git.openembedded.org/openembedded-core)

To reproduce just wait until  the language server starts and then try to edit the file. It should crash immediately

Here I tried the ``openembedded-core/meta-selftest/recipes-devtools/rust/rust-c-lib-example_git.bb``

```
[coc.nvim] Connection to server bitbake is erroring, Header must provide a Content-Length property.
{"{\"id\"":"2, \"jsonrpc\": \"2.0\", \"result\": []}Content-Length: 43","content-type":"application/vscode-jsonrpc; charset=utf-8"}. Shutting down 
server.
```

**Comment by Freed-Wu** [comment]
> Version of what

I means the version of this project, and keep oelint-adv and oelint-parser latest.

> Here I tried

It works for me.

![Screenshot from 2024-03-01 20-19-00](https://github.com/Freed-Wu/bitbake-language-server/assets/32936898/7726d099-6dee-47f2-87ca-e7f180ee6ac1)

**Comment by jclsn** [comment]
I am running Arch and the latest version of everything. Weird that it crashes for me

**Comment by jclsn** [comment]
Can you provide a way to disable the linter? That way I can at least still use the parser until I have figured out why it doesn't work

**Comment by Freed-Wu** [comment]
Can you try `paru -Syu` or `yay -Syu`?

I also use ArchLinux :smile:

**Comment by jclsn** [comment]
I just updated and restarted. It is still crashing. No idea, I will look into this next week. Something with jsonrpc. I am not good with web stuff and node

**Comment by Freed-Wu** [comment]
Tooooo weird. If you don't like it, just comment <https://github.com/Freed-Wu/bitbake-language-server/blob/main/src/bitbake_language_server/server.py#L83-L100>.

**Comment by jclsn** [comment]
Can you please provide a switch to disable the linter or parser? 

It is failing for me even with the most minimal `.vimrc`. I don't think it is only affecting me.

Don't you have any idea why the server may be missing a "Content-Length property"?

**Comment by Freed-Wu** [comment]
I think the bug is resulted from incompatiable API of old oelint-XXX. Can you show

```sh
pip show oelint-adv
pip show oelint-parser
```

**Comment by jclsn** [comment]
```bash
❯ pip show oelint-adv
Name: oelint_adv
Version: 4.4.1
Summary: Advanced bitbake-recipe linter
Home-page: https://github.com/priv-kweihmann/oelint-adv
Author: Konrad Weihmann
Author-email: kweihmann@outlook.com
License: 
Location: /usr/lib/python3.11/site-packages
Requires: anytree, colorama, oelint-parser, urllib3
Required-by: bitbake-language-server
❯ pip show oelint-parser
Name: oelint_parser
Version: 3.3.1
Summary: Alternative parser for bitbake recipes
Home-page: https://github.com/priv-kweihmann/oelint-parser
Author: Konrad Weihmann
Author-email: kweihmann@outlook.com
License: 
Location: /usr/lib/python3.11/site-packages
Requires: Deprecated, regex
Required-by: oelint_adv
```

**Comment by Freed-Wu** [comment]
> comment <https://github.com/Freed-Wu/bitbake-language-server/blob/main/src/bitbake_language_server/server.py#L83-L100>.

After comment `/usr/lib/python3.11/site-functions/bitbake-language-server/server.py`, Can it work for you?

**Comment by Freed-Wu** [comment]
Or simply, when you run `bitbake-language-server` in shell, will it crash?

**Comment by jclsn** [comment]
Haven't tried, because I don't know how to build this from source

**Comment by jclsn** [comment]
When I run it in the shell, it doesn't crash. It is as soon as I edit the file

**Comment by Freed-Wu** [comment]
> I don't know how to build this from source

```sh
sudo vi /usr/lib/python3.11/site-functions/bitbake-language-server/server.py
```

the comment 83-100.

**Comment by jclsn** [comment]
After removing those lines, it doesn't crash anymore

**Comment by Freed-Wu** [comment]
Can other features work normally?

**Comment by jclsn** [comment]
Well only the parser works then. When the linter is active and the server crashes, nothing works anymore obviously.

**Comment by jclsn** [comment]
There are similar issues [here](https://github.com/neoclide/coc-java/issues/133) and [here](https://github.com/coc-extensions/coc-omnisharp/issues/44). In the first issues the CoC developer says, that there is an issue with the language server. So I think that CoC is expecting the content-length header. Maybe you can ask Heyward Fann in the Gitter channel about it. He is very helpful

**Comment by jclsn** [comment]
Maybe you have an older CoC or NodeJS version?

**Comment by Freed-Wu** [comment]
> Well only the parser works then

Do you means hover, completion, definition goto can work except diagnostic?

> Maybe you have an older CoC or NodeJS version?

I use latest node v18.18.2 and latest coc 0.0.82. I want to wait other people can reproduce this bug and tell me how to reproduce. IMO, a bug cannot be reproduced also cannot be fixed.

**Comment by jclsn** [comment]
> Do you means hover, completion, definition goto can work except diagnostic?

Yes

>I use latest node v18.18.2 and latest coc 0.0.82. I want to wait other people can reproduce this bug and tell me how to reproduce. IMO, a bug cannot be reproduced also cannot be fixed.

I am using NodeJS 21 and tried downgrading to 18, but same error. Yeah, I understand that you need to reproduce it first.

Can you reproduce it with my vim configuration?

https://github.com/jclsn/vimconfig

**Comment by Freed-Wu** [comment]
> I want to wait other people can reproduce this bug 

In fact, I want to wait other users except coc.nvim, such as vim-lsp, nvim-lsp, even non-vim user who use other editor which support LSP, in order to judge if this bug only occur when user use coc.nvim, or it happened in every editors which support LSP. It is very important. Only when this bug only occur in coc.nvim, we can report to coc.nvim and request their helps.

**Comment by Freed-Wu** [comment]
Just be patient and everything will go well.

**Comment by jclsn** [comment]
Seems so. I just tried it with https://github.com/yegappan/lsp

and it doesn't crash

**Comment by Freed-Wu** [comment]
I believe this bug has been fixed.

The bug comes from oelint-adv will print some information about rules which
cannot be recognized by language client.

The bug only occur when wait a few seconds after enter insert mode, so it is not easy to find it. Thanks for your report!

**Comment by jclsn** [comment]
Yeah, it works now. Thank you for fixing it! :)

**Comment by jclsn** [comment]
I just had a crash again after saving the file. Not sure why that happened. It was the same error message. Couldn't yet reproduce it again though.

**Comment by jclsn** [comment]
The content-length property bug is there again since the update. The server crashes right away
